ui/src/app/intro/views/MaasIntro/MaasIntro.tsx

```diff
@@ -1,9 +1,76 @@
+import { Spinner } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+import * as Yup from "yup";
+
+import MaasIntroFields from "./MaasIntroFields";
+import type { MaasIntroValues } from "./types";
+
+import FormikForm from "app/base/components/FormikForm";
 import Section from "app/base/components/Section";
+import baseURLs from "app/base/urls";
+import machineURLs from "app/machines/urls";
+import authSelectors from "app/store/auth/selectors";
+import { actions as configActions } from "app/store/config";
+import configSelectors from "app/store/config/selectors";
+
+export const IntroSchema = Yup.object()
+  .shape({
+    name: Yup.string().required("MAAS name is required"),
+  })
+  .defined();
 
 const MaasIntro = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const authLoading = useSelector(authSelectors.loading);
+  const authUser = useSelector(authSelectors.get);
+  const errors = useSelector(configSelectors.errors);
+  const maasName = useSelector(configSelectors.maasName);
+  const configLoading = useSelector(configSelectors.loading);
+  const saved = useSelector(configSelectors.saved);
+  const saving = useSelector(configSelectors.saving);
+
```

Comment:
I think there should be a `useWindowTitle` here. Also I think we should have a `useEffect` that redirects the user if they've already completed the intro as well as redirecting on save. Lastly, should we be explicitly fetching config in this component? I suppose it's not necessary since it's fetched when the app starts.

Reply:
Thanks. I had the config being fetched here (and the auth user) but it seemed unnecessary as, like you say, it's fetched in `App`. Also, the auth user fetching was causing issues for some reason...

I think for redirecting the user if they've completed the intro we should add that later to make this easier to develop. I'll add an issue.

Reply:
Ah yeah good point! It'll much easier to only turn on redirects at the end